msbuild/Xamarin.MacDev.Tasks.Core/Tasks/CodesignTaskBase.cs

```diff
@@ -48,7 +60,19 @@ protected override string GenerateFullPathToTool ()
 			return File.Exists (path) ? path : ToolExe;
 		}
 
-		protected override string GenerateCommandLineCommands ()
+		ProcessStartInfo GetProcessStartInfo (string tool, string args)
+		{
+			var startInfo = new ProcessStartInfo (tool, args);
+
+			startInfo.WorkingDirectory = Environment.CurrentDirectory;
+			startInfo.EnvironmentVariables["CODESIGN_ALLOCATE"] = CodesignAllocate;
+
+			startInfo.CreateNoWindow = true;
+
+			return startInfo;
+		}
+
+		string GenerateCommandLineArguments (ITaskItem item)
 		{
 			var args = new ProcessArgumentBuilder ();
```

Comment:
I'm not sure what would be the **optimal** condition but, at least for `Debug` builds we should be using `--timestamp=none` to remove the network access.

It would likely be the case for other builds - except the one being published to the App Store (and likely enterprise deployment)